### PR TITLE
PTX-1940 Add timeout parameter for upgrade script

### PR DESCRIPTION
**Pull request by mborodin-px**
Signed-off-by: Maksym Borodin <maksym@portworx.com>

### CODE REVIEW

File: drivers/volume/portworx/portworx.go [hunk 1]
```diff
@@ -83,6 +83,7 @@ const (
 	startDriverTimeout               = 2 * time.Minute
 	upgradeTimeout                   = 10 * time.Minute
 	upgradeRetryInterval             = 30 * time.Second
+	upgradePerNodeTimeout            = "900"
```
Comment (thiguetta) [inline drivers/volume/portworx/portworx.go]: a small cosmetic suggestion, can we use the `time.Duration` notation as we do above and then use `duration.Seconds()` in your command?

File: drivers/volume/portworx/portworx.go [hunk 2]
```diff
@@ -356,6 +357,29 @@ func (d *portworx) getPxNode(n node.Node, cManager cluster.Cluster) (api.Node, e
...
 }
 
+func (d *portworx) getPxVersionOnNode(n node.Node, cManager cluster.Cluster) (string, error) {
+
+	pxNode, err := d.getPxNode(n, cManager)
+	if err != nil {
+		return "", err
+	}
+
+	t := func() (interface{}, bool, error) {
```
Comment (harsh-px) [inline drivers/volume/portworx/portworx.go]: this timeout look doesn't make sense.

In each retry, you are using the same pxNode that you got from a single attempt above. So nothing is going to change between subsequent retries.

File: drivers/volume/portworx/portworx.go (hunk 1 above)
Comment (mborodin-px) [reply drivers/volume/portworx/portworx.go]: fixed

File: drivers/volume/portworx/portworx.go (hunk 2 above)
Comment (mborodin-px) [reply drivers/volume/portworx/portworx.go]: fixed

File: drivers/volume/portworx/portworx.go [hunk 3]
```diff
@@ -1280,9 +1294,21 @@ func (d *portworx) UpgradeDriver(endpointURL string, endpointVersion string) err
 	}
 	logrus.Infof("permission changed on file %s", upgradeFileName)
 
+	nodeList := node.GetStorageDriverNodes()
+	pxNode := nodeList[0]
+	pxVersion, err := d.getPxVersionOnNode(pxNode, nil)
```
Comment (harsh-px) [inline drivers/volume/portworx/portworx.go]: you are ignoring the `err` here

Comment (mborodin-px) [reply drivers/volume/portworx/portworx.go]: fixed

File: drivers/volume/portworx/portworx.go [hunk 4]
```diff
@@ -356,6 +357,28 @@ func (d *portworx) getPxNode(n node.Node, cManager cluster.Cluster) (api.Node, e
 	return pxnode.(api.Node), nil
 }
 
+func (d *portworx) getPxVersionOnNode(n node.Node, cManager cluster.Cluster) (string, error) {
+
+	t := func() (interface{}, bool, error) {
+		logrus.Debugf("Getting PX Version on node [%s]", n.Name)
+		pxNode, err := d.getPxNode(n, cManager)
+		if err != nil {
+			return "", err
```
Comment (harsh-px) [inline drivers/volume/portworx/portworx.go]: don't think this compiles

**Pull request merged**